cheroot/server.py

```diff
@@ -1162,7 +1162,10 @@ def send_headers(self):  # noqa: C901  # FIXME
         # Override the decision to not close the connection if the connection
         # manager doesn't have space for it.
         if not self.close_connection:
-            can_keep = self.server.connections.can_add_keepalive_connection
+            can_keep = (
+                self.server.ready
```

Comment:
Maybe embed this into `connections.can_add_keepalive_connection` property?

Reply:
in c94eaabb400f885a93aa9bf79a7153a6d8581d41 i instead made connections a 'private' member of server, since it should never be accessed after `server.stop()` has been called.